public/components/event_analytics/utils/utils.tsx

```diff
@@ -459,3 +462,57 @@ export const getContentTabTitle = (tabID: string, tabTitle: string) => {
     </>
   );
 };
+
+/**
+ * Used to fill in missing empty data where x is an array of time values and there are only x
+ * values when y is non-zero.
+ * @param xVals all x values being used
+ * @param yVals all y values being used
+ * @param selectedInterval Moment unitOfTime used to dictate how long each interval is
+ * @param startTime starting time of x values
+ * @param endTime ending time of x values
+ * @returns an object with buckets and values where the buckets are all of the new x values and
+ * values are the corresponding values which include y values that are 0 for empty data
+ */
+export const fillTimeDataWithEmpty = (
+  xVals: string[],
+  yVals: number[],
+  selectedInterval: any,
```

Comment:
is this string? or better would be something like `'m' | 'd' | 'w' | .....`

Reply:
An example string in xVals could be "2023-01-01 00:00:00", but any datetime string can work, because it will be parsed by moment.js. So theoretically, as long as the string looks like a date, this method is useful for any time related data.

Reply:
I created a new type and put it into common/types/explorer.ts to include all possible moment.js friendly interval strings.